From f8ad3f859aadc6c81ccc667eee1439cc2e6113cf Mon Sep 17 00:00:00 2001
From: Anuj Mittal <anuj.mittal@intel.com>
Date: Tue, 30 May 2023 09:31:09 +0800
Subject: [PATCH] fix build with gcc13

* Include the headers as per:
https://gcc.gnu.org/gcc-13/porting_to.html

* Remove -Werror for:
|git/src/plugins/intel_gpu/src/kernel_selector/jitter.h:129:28: error: 'virtual kernel_selector::JitDefinitions kernel_selector::JitConstant::GetDefinitions() const' was hidden [-Werror=overloaded-virtual=]
|  129 |     virtual JitDefinitions GetDefinitions() const = 0;
|      |                            ^~~~~~~~~~~~~~

* Fix:
| /data/poky/build/tmp/work/x86-64-v3-poky-linux/openvino-inference-engine/2022.3.0-r0/git/src/inference/include/ie/ie_precision.hpp: In member function 'bool InferenceEngine::Precision::operator==(const InferenceEngine::Precision&) const':
| /data/poky/build/tmp/work/x86-64-v3-poky-linux/openvino-inference-engine/2022.3.0-r0/git/src/inference/include/ie/ie_precision.hpp:153:36: error: ambiguous overload for 'operator==' (operand types are 'const InferenceEngine::Precision::ePrecision' and 'const InferenceEngine::Precision')
|   153 |         return precisionInfo.value == p && precisionInfo.bitsSize == p.precisionInfo.bitsSize &&
|       |                ~~~~~~~~~~~~~~~~~~~ ^~ ~
|       |                              |        |
|       |                              |        const InferenceEngine::Precision
|       |                              const InferenceEngine::Precision::ePrecision
| /data/poky/build/tmp/work/x86-64-v3-poky-linux/openvino-inference-engine/2022.3.0-r0/git/src/inference/include/ie/ie_precision.hpp:152:10: note: candidate: 'bool InferenceEngine::Precision::operator==(const InferenceEngine::Precision&) const' (reversed)
|   152 |     bool operator==(const Precision& p) const noexcept {
|       |          ^~~~~~~~
| /data/poky/build/tmp/work/x86-64-v3-poky-linux/openvino-inference-engine/2022.3.0-r0/git/src/inference/include/ie/ie_precision.hpp:153:36: note: candidate: 'operator==(InferenceEngine::Precision::ePrecision, InferenceEngine::Precision::ePrecision)' (built-in)
|   153 |         return precisionInfo.value == p && precisionInfo.bitsSize == p.precisionInfo.bitsSize &&
|       |                ~~~~~~~~~~~~~~~~~~~~^~~~

Upstream-Status: Pending
Signed-off-by: Anuj Mittal <anuj.mittal@intel.com>

---
 src/inference/include/ie/ie_precision.hpp | 2 +-
 src/plugins/intel_gpu/CMakeLists.txt      | 6 +++---
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/inference/include/ie/ie_precision.hpp b/src/inference/include/ie/ie_precision.hpp
index 48bdd945b13..96c29b73159 100644
--- a/src/inference/include/ie/ie_precision.hpp
+++ b/src/inference/include/ie/ie_precision.hpp
@@ -149,7 +149,7 @@ public:
      * @return `true` if values represent the same precisions, `false` otherwise
      */
     bool operator==(const Precision& p) const noexcept {
-        return precisionInfo.value == p && precisionInfo.bitsSize == p.precisionInfo.bitsSize &&
+        return precisionInfo.value == p.precisionInfo.value && precisionInfo.bitsSize == p.precisionInfo.bitsSize &&
                areSameStrings(precisionInfo.name, p.precisionInfo.name);
     }
 
diff --git a/src/plugins/intel_gpu/CMakeLists.txt b/src/plugins/intel_gpu/CMakeLists.txt
index 948aecfcfec..a1b9783068b 100644
--- a/src/plugins/intel_gpu/CMakeLists.txt
+++ b/src/plugins/intel_gpu/CMakeLists.txt
@@ -36,9 +36,9 @@ set(INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/include")
 
 add_subdirectory(thirdparty)
 
-if(CMAKE_COMPILER_IS_GNUCXX)
-    ie_add_compiler_flags(-Werror)
-endif()
+#if(CMAKE_COMPILER_IS_GNUCXX)
+#    ie_add_compiler_flags(-Werror)
+#endif()
 
 add_subdirectory(src/runtime)
 add_subdirectory(src/kernel_selector)
-- 
2.34.1

